presto-geospatial/src/test/java/com/facebook/presto/geospatial/BenchmarkGeometryToBingTiles.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.geospatial;
+
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.VerboseMode;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+@State(Scope.Thread)
+@Fork(2)
+@Warmup(iterations = 5, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@Measurement(iterations = 5, time = 500, timeUnit = TimeUnit.MILLISECONDS)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@BenchmarkMode(Mode.AverageTime)
+public class BenchmarkGeometryToBingTiles
+{
+    @Benchmark
+    public Object geometryToBingTiles(BenchmarkData data)
```

Comment:
Add `BlackHole blackHole` to the method parameters and do `blackHole.consume(BingTileFunctions.geometryToBingTiles(data.geometry, data.zoomLevel));` so that JVM doesn't optimize things away seeing that the returned object is not used at all.

Same for the `envelopeToBingTiles` call below.

Reply:
It's possible that jmh may be taking care of this when we return non-void, but I am not sure. So, it is better to make sure we don't have dead code elimination.

Reply:
@nezihyigitbasi , http://java-performance.info/jmh/ suggests that returning a value is as good as using BlackHole and recommends using BlackHole for cases when multiple values need to be returned.

```
Dead code

Dead code elimination is a well known problem among microbenchmark writers. The general solution is to use the result of calculations somehow. JMH does not do any magic tricks on its own. If you want to defend against dead code elimination – never write void tests. Always return the result of your calculations. JMH will take care of the rest.
```

Let me know if you still like me to change the benchmark to use the BlackHole.

Reply:
Did you try running the benchmark with the blackhole version to see whether there is any difference in the number? If the numbers are the same then we can keep the code as is.

Reply:
@nezihyigitbasi , I ran 3 versions of this benchmark. Here are the results:

as-is

Benchmark                                         Mode  Cnt    Score   Error  Units
BenchmarkGeometryToBingTiles.envelopeToBingTiles  avgt   10    0.023 ± 0.011  ms/op
BenchmarkGeometryToBingTiles.geometryToBingTiles  avgt   10  319.748 ± 5.753  ms/op

void methods

Benchmark                                         Mode  Cnt    Score    Error  Units
BenchmarkGeometryToBingTiles.envelopeToBingTiles  avgt   10    0.025 ±  0.017  ms/op
BenchmarkGeometryToBingTiles.geometryToBingTiles  avgt   10  310.727 ± 23.516  ms/op

with BlackHole

Benchmark                                         Mode  Cnt    Score    Error  Units
BenchmarkGeometryToBingTiles.envelopeToBingTiles  avgt   10    0.023 ±  0.009  ms/op
BenchmarkGeometryToBingTiles.geometryToBingTiles  avgt   10  309.477 ± 31.728  ms/op